File: KRC-PGR/pgr-frontend/src/constants/Authority.ts
export enum AuthorityID {
    Admin = 0,
    Manager = 1,
    User = 2
}

export enum AuthorityName {
    Admin = 'admin',
    Manager = 'manager',
    User = 'user'
}

export enum AuthorityJapaneseName {
    Admin = '管理者',
    Manager = '運営者',
    User = 'ユーザ'
}

export class Authority {
    public static Admin = {
        id: AuthorityID.Admin,
        name: AuthorityName.Admin,
        ja: AuthorityJapaneseName.Admin,
    }
    public static Manager = {
        id: AuthorityID.Manager,
        name: AuthorityName.Manager,
        ja: AuthorityJapaneseName.Manager,
    }
    public static User = {
        id: AuthorityID.User,
        name: AuthorityName.User,
        ja: AuthorityJapaneseName.User,
    }
}

File: KRC-PGR/pgr-backend/src/main/java/com/krc/pgr/constant/Authority.java
package com.krc.pgr.constant;

public enum Authority {
    GUEST(-1), USER(2), MANAGER(1), ADMIN(0);

    private int authority_id;

    Authority(int authority_id) {
        this.authority_id = authority_id;
    }

    public int getAuthority() {
        return this.authority_id;
    }
}
